File: src/main/java/io/vertigo/shiny/components/dataviz/gauge/ShinyGauge.java
package io.vertigo.shiny.components.dataviz.gauge;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyGauge(
		String title,
		double value,
		double maxValue) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinyGauge";
	}

	public ShinyGauge {
		Assertion.check().isNotBlank(title, "Title cannot be blank");
	}
}

File: banshee/src/models/ShinyGauge.ts
import { ShinyComponent } from '../ShinyComponent';

export class ShinyGauge implements ShinyComponent {
  title: string;
  value: number;
  maxValue: number;
  type: string = 'gauge';

  constructor(title: string, value: number, maxValue: number) {
    this.title = title;
    this.value = value;
    this.maxValue = maxValue;
  }
}
